controller/src/main/java/org/jboss/as/controller/operations/global/ReadAttributeHandler.java

```diff
@@ -281,9 +281,19 @@ static void resolveAttribute(OperationContext context, AttributeDefinition attri
             final ModelNode result = subModel.get(attribute.getName());
             context.getResult().set(result);
         } else if (defaults && attribute.getDefaultValue() != null) {
-            // No defined value in the model. See if we should reply with a default from the metadata,
-            // reply with undefined, or fail because it's a non-existent attribute name
-            context.getResult().set(attribute.getDefaultValue());
+            if(!attribute.hasAlternative(subModel)) {
```

Comment:
Yes you are right, it is an enhancement. I started backporting this issue, because on 7.2 it is reported as bug - https://issues.jboss.org/browse/JBEAP-7392 and I missed that on WF it is an enhancement. In that case backporting doesn't make sense.